beancount.el

```diff
@@ -448,6 +458,27 @@ With an argument move to the next non cleared transaction."
   "A list of the accounts available in this buffer.")
 (make-variable-buffer-local 'beancount-accounts)
 
+(defun beancount-inside-string-p ()
+  (nth 3 (syntax-ppss)))
+
+(defun beancount-looking-at-tag-generic (start-char start-char-string)
+  (save-excursion
+    (unless (beancount-inside-string-p)
+      ;; Loop backward until the previous character is not a valid tag
+      ;; character
+      (while (char-table-range beancount-tag-char-table (char-before))
+        (backward-char))
+      ;; Move before the start character if we hit it
+      (when (eq (char-before) start-char)
+        (backward-char))
+      (looking-at (concat start-char-string "[" beancount-tag-chars "]*")))))
```

Comment:
Cannot this based on `(re-search-backward (concat "[^" beancount-tag-chars "]"))` instead?

Reply:
Yes, that is a better solution.

I was so caught up in a "don't use `looking-backward`" mindset that I seemed to have mentally excluded any sort of backwards regexp search and way overthought the solution.